e2e-tests/gef/cypress/e2e/clone/facility-end-date-enabled/clone-v0-to-v1.spec.js

```diff
@@ -0,0 +1,110 @@
+import relative from '../../relativeURL';
+import { BANK1_MAKER1 } from '../../../../../e2e-fixtures/portal-users.fixture';
+import cloneGEFDeal from '../../pages/clone-deal';
+import mandatoryCriteria from '../../pages/mandatory-criteria';
+import nameApplication from '../../pages/name-application';
+import applicationDetails from '../../pages/application-details';
+import facilityEndDate from '../../pages/facility-end-date';
+import bankReviewDate from '../../pages/bank-review-date';
+import facilities from '../../pages/facilities';
+import { submitButton } from '../../partials';
+import aboutFacilityUnissued from '../../pages/unissued-facilities-about-facility';
+
+/**
+ * NOTE: These tests check the backwards compatibility with in-flight version 0 deals.
+ * If this test is modified then version 0 deals may no longer be supported and
+ * we need to update these deals in production.
+ */
+context('Clone version 0 deal to version 1', () => {
+  let version0DealId;
+  let clonedDealId;
+
+  before(() => {
+    cy.insertVersion0Deal(BANK1_MAKER1.username).then(({ insertedId: dealId }) => {
+      version0DealId = dealId;
+    });
+  });
+
+  beforeEach(() => {
+    cy.saveSession();
+  });
+
+  describe('cloning a version 0 deal', () => {
+    before(() => {
+      cy.login(BANK1_MAKER1);
+      cy.visit(relative(`/gef/application-details/${version0DealId}`));
+
+      applicationDetails.addContingentFacilityButton().click();
+      facilities.hasBeenIssuedRadioYesRadioButton().click();
+      cy.clickContinueButton();
+
+      cy.visit(relative(`/gef/application-details/${version0DealId}`));
+
+      cloneGEFDeal.cloneGefDealLink().click();
+      mandatoryCriteria.trueRadio().click();
+
+      cy.clickContinueButton();
+
+      cy.keyboardInput(nameApplication.internalRef(), 'Cloned of v0 deal');
+      cy.clickContinueButton();
+
+      cy.get('table.govuk-table tr').eq(1).find('td').eq(1).find('.govuk-link').click();
```

Comment:
What link is this, are we able to use a selector instead? Looks like this was added in `clone-gef-deal.spec.js` and then copied over to these new E2E tests

Reply:
I copied this across & couldnt find a good alternative. Its pretty niche and only needed when cloning a deal.

Basically this gets the first deal from the deals page. In portal tests there's some additional selectors (`e2e-tests\portal\cypress\e2e\pages\dashboardDeals.js`) but you need to know the id in advance - e.g.
```
    link: (id) => cy.get(`[data-cy="deal__link--${id}"]`),
```

I could add something to the sparse `dashboad.js` page in gef
```
  facility: (row) => ({
    link: () => cy.get('table.govuk-table tr').eq(1).find('td').eq(row).find('.govuk-link').click(),
  }),
```
It probably needs some tweaking & it still has to use the slightly ugly `.eq` and `.find` syntax because the data-cy selector contains the deal id.

Reply:
Thanks @MarRobSoftwire 

It looks like this needs refactoring completely. We should have something like this

```js
table: {
  body: {
    row: (dealId) => ({
      status: () => cy.get(`[data-cy="ref-${dealId}-status"]`),
    });
  },
},
```

Then tests can just do this 

```
table.body.row(dealId).status().click();
```

With `dealId` and/or `index`

Not for this PR though, this could break a lot of things. Also, this could conflict with an upcoming PR regarding some E2E refactoring for GEF deals. Resolving